src/controllers/authController.ts

```diff
@@ -93,5 +95,61 @@ const login = async (req: Request, res: Response): Promise<void> => {
     sendInternalErrorResponse(res, err);
   }
 };
+const updatePassword = async (req: Request, res: Response): Promise<void> => {
+  try {
+    const { oldPassword, newPassword } = req.body;
+    const token = req.headers.authorization?.split(' ')[1];
+    if (!token) {
+      res.status(401).json({
+        ok: false,
+        message: 'Unauthorized',
+      });
+      return;
+    }
+    const decode = jwt.verify(token, process.env.JWT_SECRET as string) as { id: string };
+    if (!decode) {
+      res.status(400).json({
+        ok: false,
+        message: 'Invalid token',
+      });
+    }
```

Comment:
I guess you aren't supposed to write this particular code cuz there's a reusable function in middleware folder for that.